site_scons/site_tools/go_builder.py

```diff
@@ -49,6 +49,28 @@ def _scan_go_file(node, env, _path):
     return includes
 
 
+def get_go_version(output):
+    """Convert from a "go version" string to a well-formatted major.minor.point string"""
+    go_version = output.split(' ')[2].replace('go', '')
+    if '-' in go_version:
+        go_version = go_version.split('-')[0]
+    parts = []
+    for part in go_version.split("."):
+        if 'rc' in part:
+            parts.append(part.split("rc", maxsplit=1)[0])
+        else:
+            parts.append(part)
+    return ".".join(parts)
```

Comment:
I know you're not usually a fan of regex, but...
```
def get_go_version(output):
    # Capture only the number, ignoring optional rc
    one_part = '([0-9]+)(?:rc[0-9]+)?'
    return '.'.join(re.findall(fr'(?:go)?{one_part}\.{one_part}\.{one_part}', output)[0])
```

Reply:
I'll try it out once the current test has completed and I can add a test for the actual output that we're actually parsing here.